tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -352,6 +353,106 @@ def test_5_inc_snapshots(network_config,
     test_matrix.run_test(_test_seq_snapshots)
 
 
+DEFAULT_HOST_IP = "192.168.0.1"
+DEFAULT_GUEST_IP = "192.168.0.2"
+DEFAULT_TAP_NAME = "tap0"
+DEFAULT_DEV_NAME = "eth0"
+DEFAULT_NETMASK = 30
+
+
+def _get_process_start_time(p_pid):
+    # Helper function - returns the start_time and returncode for
+    # process with provided PID
+    proc = subprocess.Popen(["stat -c%X /proc/{}".format(p_pid)],
+                            stdout=subprocess.PIPE,
+                            shell=True,
+                            encoding='ascii')
+    out = proc.communicate()[0].strip('\n')
+    return out, proc.returncode
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_load_snapshot_failure_handling(bin_cloner_path):
+    """
+    Test scenario.
+
+    1.  Create a new snapshot from a Firecracker process.
+    2.  Corrupt the snapshot's memory file.
+    3.  Try to build a VM out of the corrupted snapshot.
+    4.  Verify that an error was shown and the process is dead.
+    """
+    logger = logging.getLogger("snapshot_load_failure")
+
+    # Use a predefined vm instance.
+    vm_instance = VMNano.spawn(bin_cloner_path)
+    vm = vm_instance.vm
+    root_disk = vm_instance.disks[0]
+    ssh_key = vm_instance.ssh_key
+
+    vm.start()
+    ssh_connection = net_tools.SSHConnection(vm.ssh_config)
+
+    # Verify if guest can run commands.
+    exit_code, _, _ = ssh_connection.execute_command("sync")
+    assert exit_code == 0
+
+    logger.info("Create %s.", SnapshotType.FULL)
+    # Create a snapshot builder from a microvm.
+    snapshot_builder = SnapshotBuilder(vm)
+
+    disks = [root_disk.local_path()]
+    # Create base snapshot.
+    snapshot = snapshot_builder.create(disks,
+                                       ssh_key,
+                                       SnapshotType.FULL)
+    vm.kill()
+
+    # Corrupt the memory file of the snapshot
+    logger.info("Corrupt the memory file %s", snapshot.mem)
+
+    mem_fd = open(snapshot.mem, "wb")
+    mem_fd.truncate(0)
+    mem_fd.close()
+
+    # Create a new VM instance
+    vm = MicrovmBuilder(bin_cloner_path).create_basevm()
+    vm.spawn(log_level='Info')
+
+    # Hardlink all the snapshot files into the microvm jail.
+    jailed_mem = vm.create_jailed_resource(snapshot.mem)
+    jailed_vmstate = vm.create_jailed_resource(snapshot.vmstate)
```

Comment:
Can you achieve the same without the first half of this test?

Instead of actually starting a base vm, creating a snapshot and corrupting that snapshot - just try to load empty `jailed_mem` and empty `jailed_vmstate`.

That would significantly cut both complexity and duration of this test.

Reply:
Done